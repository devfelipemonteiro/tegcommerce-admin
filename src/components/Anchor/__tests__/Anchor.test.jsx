import React from 'react';
import { shallow } from 'enzyme';
import Anchor from '../Anchor';

const minimalProps = {
  children: <div>Link</div>
};

const commonProps = {
  ...minimalProps,
  href: 'http://www.google.com',
  title: 'Anchor title',
  onClick: jest.fn()
};

describe('<Anchor /> Tests', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render without exploding', done => {
    const wrapper = shallow(<Anchor {...minimalProps} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<Anchor {...minimalProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with all props', done => {
    const wrapper = shallow(<Anchor {...commonProps}></Anchor>);
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should call onClick handler when click on anchor', done => {
    const wrapper = shallow(<Anchor {...commonProps} />);
    const anchor = wrapper
      .dive()
      .find('a')
      .first();
    anchor.simulate('click');
    expect(commonProps.onClick.mock.calls.length).toBe(1);
    done();
  });
});
