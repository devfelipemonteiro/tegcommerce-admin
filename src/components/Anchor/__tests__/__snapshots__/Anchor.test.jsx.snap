// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> Tests should render correctly 1`] = `
<Anchor
  href="#"
  style={null}
>
  <div>
    Link
  </div>
</Anchor>
`;

exports[`<Anchor /> Tests should render correctly with all props 1`] = `
<Anchor
  href="http://www.google.com"
  onClick={[MockFunction]}
  style={null}
  title="Anchor title"
>
  <div>
    Link
  </div>
</Anchor>
`;
